src/Bridge/Symfony/DependencyInjection/Compiler/AdapterCompilerPass.php

```diff
@@ -36,10 +36,12 @@ public function process(ContainerBuilder $container): void
             $container->removeDefinition('sonata.doctrine.adapter.doctrine_orm');
         }
 
-        if ($container->has('doctrine_phpcr')) {
-            $definition->addMethodCall('addAdapter', [new Reference('sonata.doctrine.adapter.doctrine_phpcr')]);
-        } else {
-            $container->removeDefinition('sonata.doctrine.adapter.doctrine_phpcr');
+        if ($container->has('sonata.doctrine.adapter.doctrine_phpcr')) {
+            if ($container->has('doctrine_phpcr')) {
+                $definition->addMethodCall('addAdapter', [new Reference('sonata.doctrine.adapter.doctrine_phpcr')]);
+            } else {
+                $container->removeDefinition('sonata.doctrine.adapter.doctrine_phpcr');
+            }
         }
```

Comment:
```suggestion
        if ($container->has('sonata.doctrine.adapter.doctrine_phpcr')) {
            $definition->addMethodCall('addAdapter', [new Reference('sonata.doctrine.adapter.doctrine_phpcr')]);
        }
```
Then in `SonataDoctrineExtension` change 
```php
if (class_exists(DocumentManager::class)) {
    $loader->load('doctrine_phpcr.xml');
}
```
to:
```php
$bundles = $container->getParameter('kernel.bundles');

if (class_exists(DocumentManager::class) && isset($bundles['DoctrinePHPCRBundle'])) {
    $loader->load('doctrine_phpcr.xml');
}
```

This will also resolve problem with no existing argument in `sonata.doctrine.adapter.doctrine_phpcr`.

Reply:
Should the SonataDoctrineExtension not look like this:

```php
        $bundles = $container->getParameter('kernel.bundles');

        if (interface_exists(EntityManagerInterface::class) && isset($bundles['SonataDoctrineORMAdminBundle'])) {
            $loader->load('doctrine_orm.xml');
            $loader->load('mapper_orm.xml');
        }

        if (class_exists(DocumentManager::class) && isset($bundles['SonataDoctrinePHPCRAdminBundle'])) {
            $loader->load('doctrine_phpcr.xml');
        }
```

to check the bundle is there and not only a doctrine bundle

Reply:
@MaxE17677
No. In `ORM` SonataDoctrineORMAdminBundle is not require. 
```
private function isDoctrineOrmLoaded(ContainerBuilder $container): bool
    {
        return $container->has('doctrine') && $container->has('sonata.doctrine.adapter.doctrine_orm');
    }
```
```

        if (interface_exists(Doctrine\ORM\EntityManagerInterface::class)) {
            $loader->load('doctrine_orm.xml');
            $loader->load('mapper_orm.xml');
        }
```
```
<service id="sonata.doctrine.adapter.doctrine_orm" class="Sonata\Doctrine\Adapter\ORM\DoctrineORMAdapter" public="false">
            <argument type="service" id="doctrine"/>
        </service>
```

Reply:
Also in `DoctrinePHPCRBundle` is checked, not `SonataDoctrinePHPCRAdminBundle`. This both bundles are related to admin(allow use database system with admin). They are not related to `SonataDoctrineBundle`.